### PR TITLE
new_module: backup_restore_job_info

### PR DESCRIPTION
**Pull request by mandar242**
##### SUMMARY

Adding a new module for getting restore job info.

Uses following boto3 APIs
- [list_restore_jobs](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/backup/client/list_restore_jobs.html)
- [describe_restore_job](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/backup/client/describe_restore_job.html)

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

backup_restore_job_info
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by mandar242** [comment]
recheck

File: plugins/modules/backup_restore_job_info.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    - List detailed information about AWS Backup restore jobs initiated to restore a saved resource.
+author:
+    - Mandar Vijay Kulkarni (@mandar242)
+options:
+  account_id:
+    description:
+      - The account ID to list the restore jobs from.
+    required: false
+    type: str
+  status:
```
Comment (abikouo) [inline plugins/modules/backup_restore_job_info.py:22]: what if you define it as a list of str with a set of choices?

File: plugins/modules/backup_restore_job_info.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    description:
+      - Status of restore jobs to filter the result based on job status.
+    required: false
+    choices: ['PENDING', 'RUNNING', 'COMPLETED', 'ABORTED', 'FAILED']
+    type: str
+  created_before:
```
Comment (abikouo) [inline plugins/modules/backup_restore_job_info.py:28]: please provide an example or a definition of the date format (same for other date parameters)
Even though it is part of the example section, could be worth adding it here

File: plugins/modules/backup_restore_job_info.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+# -*- coding: utf-8 -*-
+
+# Copyright: Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+---
+module: backup_restore_job_info
+version_added: 6.0.0
+short_description: List information about backup restore jobs.
```
Comment (alinabuzachis) [inline plugins/modules/backup_restore_job_info.py:11]:
```suggestion
short_description: List information about backup restore jobs
```

File: plugins/modules/backup_restore_job_info.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+
+- name: List restore jobs based on Account ID
+  amazon.aws.backup_restore_job_info:
+    account_id: xx1234567890
+
+- name: List restore jobs based on status and created_before time
+  amazon.aws.backup_restore_job_info:
+    status: completed
+    created_before: "2023-02-25T00:05:36.309Z"
+"""
```
Comment (alinabuzachis) [inline plugins/modules/backup_restore_job_info.py:72]: I guess this module should also support to gather information on a specific restore job also (https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/backup/client/describe_restore_job.html) and not only listing jobs.

@tremble what do you think?

Comment (tremble) [reply plugins/modules/backup_restore_job_info.py:72]: +1 describing is generally useful functionality

File: plugins/modules/backup_restore_job_info.py [hunk 5]
```diff
@@ -0,0 +1,211 @@
...
+
+def list_restore_jobs(connection, module, request_args):
+    try:
+        response = _list_restore_jobs(connection, **request_args)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg="Failed to list restore jobs")
+
+    snaked_restore_jobs = [camel_dict_to_snake_dict(restore_job) for restore_job in response["RestoreJobs"]]
+
+    return snaked_restore_jobs
```
[the remark this answers is not in the document]
Comment (mandar242) [reply plugins/modules/backup_restore_job_info.py]: Fixed.

File: plugins/modules/backup_restore_job_info.py (hunk 3 above)
Comment (mandar242) [reply plugins/modules/backup_restore_job_info.py]: Fixed.

**Review by gravesm — APPROVED** [review]

**Pull request merged**